### DISCUSSION TITLE
Code developed to Integrate a Gradient on a Surface

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#28543] [Q&A Modules: General] [answered]
I have modified some code to determine the value of the surface integral pictured. The values that are being calculated from the code as developed are not in total agreement with the reference value. Would you be as so kind to review what I have modified and point out the potential errors that I have that may be causing my errors?

[SideDiffusiveFluxDotNormIntegralDotComp_crab_C.txt](https://github.com/user-attachments/files/16883894/SideDiffusiveFluxDotNormIntegralDotComp_crab_C.txt)
[SideDiffusiveFluxDotNormIntegralDotComp_crab_h.txt](https://github.com/user-attachments/files/16883895/SideDiffusiveFluxDotNormIntegralDotComp_crab_h.txt)

![image](https://github.com/user-attachments/assets/6fca347e-c3b2-41eb-96b5-0582d348ed9c)

**Comment by GiudGiud** [comment]
Hello

There's a lot in this expression. is u_tau the tangeantial component to a vector field, tangeant to the surface? and ns the normal? what is n_y ?

**Reply by TLWise** [reply]
Thank you for taking a look at it. You are correct, n is the normal, n_y is the normal to the y-axis. I will have to verify u_tau, I believe it is the u component tangential to the surface, where its value may be zero at the surface, but its derivative is not necessarily zero.

**Reply by GiudGiud** [reply, marked as answer]
I would recommend you remove the "diffusivity gradient product" routines, and write down the expression directly

**Reply by TLWise** [reply]
Would you be able to provide a simple example of writing down an expression directly that involves a derivative with respect to the normal? I'm not sure of how to write it in the syntax required by MOOSE.

**Reply by GiudGiud** [reply]
sure, use the chain rule
```
du/dn = du/dx dx/dn + du/dy dy/dn ...
```
then dx/dn is 1/n_x
